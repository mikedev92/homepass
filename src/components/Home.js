// @flow
import React from 'react';
import { Dimensions, Image, TouchableOpacity } from 'react-native';
import { ListItem, Container, Content, Text, Button, View, Thumbnail, Body, Right, Col, Left, CardItem, Icon, Badge } from 'native-base';
import gql from 'graphql-tag';
import { get, upperCase, range } from 'lodash';
import { graphql } from 'react-apollo';
import { compose } from 'recompose';
import moment from 'moment';
import { withNavigation } from 'react-navigation';
import { graphqlLoader, Feed, Carousel } from './common';

type Props = {
	data: ApolloData<*>,
	navigation: Navigation
};

function getHomeText(numPlaces: number) {
	if (numPlaces > 1) {
		return `You're selling ${numPlaces} homes. It's a beautiful day ☀️`;
	}
	if (numPlaces === 1) {
		return "You're selling 1 home. It's a beautiful day ☀️";
	}
	return "You're selling no homes today, relax enjoy the sun ☀️";
}

function Home({ data, navigation }: Props) {
	const numPlaces = get(data, 'viewer.spaces.edges', []).length;
	return (
		<Container>
			<Content statusBarPadding>
				<ListItem noBorder>
					<Col>
						<View>
							<Text status>
								{upperCase(moment().format('dddd, DD MMMM'))}
							</Text>
							{data.viewer && data.viewer.firstName
								? <Text display>
										Hi, {data.viewer.firstName}!
									</Text>
								: <Text display>Hey there</Text>}
						</View>
					</Col>
					<Right style={{ paddingBottom: 8, alignSelf: 'flex-end' }}>
						<TouchableOpacity transparent onPress={() => navigation.navigate('Profile')}>
							{data.viewer && data.viewer.imageUrl
								? <Thumbnail source={{ uri: data.viewer.imageUrl }} />
								: <Badge>
										<Text>Profile</Text>
									</Badge>}
						</TouchableOpacity>
					</Right>
				</ListItem>
				<ListItem noBorder>
					<Text accent>
						{getHomeText(numPlaces)}
					</Text>
				</ListItem>
				{numPlaces > 0 &&
					<ListItem itemDivider>
						<Body>
							<Text>Places</Text>
						</Body>
						<Right>
							<Button onPress={() => navigation.navigate('SpaceList')} transparent noPadding>
								<Text>See all</Text>
							</Button>
						</Right>
					</ListItem>}
				{numPlaces > 0 &&
					<ListItem noBorder style={{ marginLeft: 0, paddingRight: 0 }}>
						<Carousel>
							{get(data, 'viewer.spaces.edges', []).map((edge: SpaceEdge) =>
								<TouchableOpacity key={get(edge, 'node.id')} style={{ width: Dimensions.get('window').width - 20, paddingHorizontal: 5 }} onPress={() => navigation.navigate('Space', { id: get(edge, 'node.id') })} activeOpacity={1}>
									<CardItem cardBody>
										<Image style={{ width: '100%', height: 170, borderRadius: 4 }} source={get(edge, 'node.imageUrl') && { uri: get(edge, 'node.imageUrl') }} />
									</CardItem>
									<CardItem cardBody onPress={() => navigation.navigate('Space', { id: get(edge, 'node.id') })}>
										<Col>
											<Text style={{ marginTop: 4 }} body>
												{get(edge, 'node.title')}
											</Text>

											{get(edge, 'node.nextEvent') &&
												<Text rt style={{ marginTop: 0 }} status alert>
													{upperCase(`${get(edge, 'node.nextEvent.name')} ${moment(get(edge, 'node.nextEvent.startDate')).fromNow()}`)}
												</Text>}
										</Col>
									</CardItem>
								</TouchableOpacity>
							)}
						</Carousel>
					</ListItem>}
				<ListItem itemDivider>
					<Body>
						<Text>Activity</Text>
					</Body>
				</ListItem>
				<ListItem noBorder>
					<Feed>
						{range(10).map(i =>
							<ListItem key={i}>
								<Left>
									<Icon name="arrow-back" />
								</Left>
								<Body>
									<Text>Note added to christina for 27 Elizabeth St</Text>
									<Text status>Today at 9:41AM</Text>
									{i % 4 === 0 ? <Text>Some extra content</Text> : null}
								</Body>
								<Right>
									<Thumbnail source={{ uri: 'https://unsplash.it/200/300' }} />
								</Right>
							</ListItem>
						)}
					</Feed>
				</ListItem>
			</Content>
		</Container>
	);
}

const HomeQuery = gql`
	query HomeQuery {
		viewer {
			firstName
			imageUrl
			spaces {
				edges {
					node {
						id
						imageUrl
						title
						nextEvent {
							name
							startDate
						}
					}
				}
			}
		}
	}
`;

export default compose(graphql(HomeQuery), withNavigation, graphqlLoader)(Home);
