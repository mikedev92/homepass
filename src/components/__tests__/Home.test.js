// @flow
import React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate';
import Home from '../Home';

const testProps = [
	{
		data: {
			viewer: {
				id: '234',
				imageUrl: 'somePictureUrl',
				firstName: 'First Name',
				spaces: {
					edges: []
				}
			}
		}
	},
	{
		data: {
			viewer: {
				spaces: {
					edges: [
						{
							node: {
								id: '123',
								title: 'Property',
								nextEvent: {
									name: 'Woo',
									startDate: new Date(0),
									endDate: new Date(1000000000000000000)
								}
							}
						}
					]
				}
			}
		}
	},
	{
		data: {
			viewer: {
				id: '234',
				imageUrl: 'somePictureUrl',
				firstName: 'First Name',
				spaces: {
					edges: [
						{
							node: {
								id: '123',
								title: 'Property 1',
								nextEvent: {
									name: 'Woo',
									startDate: new Date(0),
									endDate: new Date(1000000000000000000)
								}
							}
						},
						{
							node: {
								id: '1234',
								title: 'Property 2',
								nextEvent: {
									name: 'Woo',
									startDate: new Date(0),
									endDate: new Date(1000000000000000000)
								}
							}
						}
					]
				}
			}
		}
	}
];

describe('<Home />', () => {
	beforeEach(() => {
		MockDate.set(new Date('1/1/2000'));
	});

	afterEach(() => {
		MockDate.reset();
	});

	it('matches snapshots', () => {
		testProps.forEach(props => {
			const wrapper = shallow(<Home {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
