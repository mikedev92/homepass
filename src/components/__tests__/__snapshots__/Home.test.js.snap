// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshots 1`] = `
<Styled(Container)>
  <Styled(Content)
    statusBarPadding={true}
  >
    <Styled(ListItem)
      noBorder={true}
    >
      <ColumnNB>
        <Styled(ViewNB)>
          <Styled(Text)
            status={true}
          >
            SATURDAY 01 JANUARY
          </Styled(Text)>
          <Styled(Text)
            display={true}
          >
            Hi, 
            First Name
            !
          </Styled(Text)>
        </Styled(ViewNB)>
      </ColumnNB>
      <Styled(Right)
        style={
          Object {
            "alignSelf": "flex-end",
            "paddingBottom": 8,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Thumbnail)
            source={
              Object {
                "uri": "somePictureUrl",
              }
            }
          />
        </TouchableOpacity>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
    >
      <Styled(Text)
        accent={true}
      >
        You're selling no homes today, relax enjoy the sun ☀️
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Activity
        </Styled(Text)>
      </Styled(Body)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
    >
      <Styled(Feed)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(Feed)>
    </Styled(ListItem)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`<Home /> matches snapshots 2`] = `
<Styled(Container)>
  <Styled(Content)
    statusBarPadding={true}
  >
    <Styled(ListItem)
      noBorder={true}
    >
      <ColumnNB>
        <Styled(ViewNB)>
          <Styled(Text)
            status={true}
          >
            SATURDAY 01 JANUARY
          </Styled(Text)>
          <Styled(Text)
            display={true}
          >
            Hey there
          </Styled(Text)>
        </Styled(ViewNB)>
      </ColumnNB>
      <Styled(Right)
        style={
          Object {
            "alignSelf": "flex-end",
            "paddingBottom": 8,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Badge)>
            <Styled(Text)>
              Profile
            </Styled(Text)>
          </Styled(Badge)>
        </TouchableOpacity>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
    >
      <Styled(Text)
        accent={true}
      >
        You're selling 1 home. It's a beautiful day ☀️
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Places
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(Button)
          noPadding={true}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Text)>
            See all
          </Styled(Text)>
        </Styled(Button)>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
      style={
        Object {
          "marginLeft": 0,
          "paddingRight": 0,
        }
      }
    >
      <HomepassCarousel>
        <TouchableOpacity
          activeOpacity={1}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "paddingHorizontal": 5,
              "width": 730,
            }
          }
        >
          <Styled(CardItem)
            cardBody={true}
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 170,
                  "width": "100%",
                }
              }
            />
          </Styled(CardItem)>
          <Styled(CardItem)
            cardBody={true}
            onPress={[Function]}
          >
            <ColumnNB>
              <Styled(Text)
                body={true}
                style={
                  Object {
                    "marginTop": 4,
                  }
                }
              >
                Property
              </Styled(Text)>
              <Styled(Text)
                alert={true}
                rt={true}
                status={true}
                style={
                  Object {
                    "marginTop": 0,
                  }
                }
              >
                WOO 30 YEARS AGO
              </Styled(Text)>
            </ColumnNB>
          </Styled(CardItem)>
        </TouchableOpacity>
      </HomepassCarousel>
    </Styled(ListItem)>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Activity
        </Styled(Text)>
      </Styled(Body)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
    >
      <Styled(Feed)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(Feed)>
    </Styled(ListItem)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`<Home /> matches snapshots 3`] = `
<Styled(Container)>
  <Styled(Content)
    statusBarPadding={true}
  >
    <Styled(ListItem)
      noBorder={true}
    >
      <ColumnNB>
        <Styled(ViewNB)>
          <Styled(Text)
            status={true}
          >
            SATURDAY 01 JANUARY
          </Styled(Text)>
          <Styled(Text)
            display={true}
          >
            Hi, 
            First Name
            !
          </Styled(Text)>
        </Styled(ViewNB)>
      </ColumnNB>
      <Styled(Right)
        style={
          Object {
            "alignSelf": "flex-end",
            "paddingBottom": 8,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Thumbnail)
            source={
              Object {
                "uri": "somePictureUrl",
              }
            }
          />
        </TouchableOpacity>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
    >
      <Styled(Text)
        accent={true}
      >
        You're selling 2 homes. It's a beautiful day ☀️
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Places
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(Button)
          noPadding={true}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Text)>
            See all
          </Styled(Text)>
        </Styled(Button)>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
      style={
        Object {
          "marginLeft": 0,
          "paddingRight": 0,
        }
      }
    >
      <HomepassCarousel>
        <TouchableOpacity
          activeOpacity={1}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "paddingHorizontal": 5,
              "width": 730,
            }
          }
        >
          <Styled(CardItem)
            cardBody={true}
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 170,
                  "width": "100%",
                }
              }
            />
          </Styled(CardItem)>
          <Styled(CardItem)
            cardBody={true}
            onPress={[Function]}
          >
            <ColumnNB>
              <Styled(Text)
                body={true}
                style={
                  Object {
                    "marginTop": 4,
                  }
                }
              >
                Property 1
              </Styled(Text)>
              <Styled(Text)
                alert={true}
                rt={true}
                status={true}
                style={
                  Object {
                    "marginTop": 0,
                  }
                }
              >
                WOO 30 YEARS AGO
              </Styled(Text)>
            </ColumnNB>
          </Styled(CardItem)>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={1}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "paddingHorizontal": 5,
              "width": 730,
            }
          }
        >
          <Styled(CardItem)
            cardBody={true}
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 170,
                  "width": "100%",
                }
              }
            />
          </Styled(CardItem)>
          <Styled(CardItem)
            cardBody={true}
            onPress={[Function]}
          >
            <ColumnNB>
              <Styled(Text)
                body={true}
                style={
                  Object {
                    "marginTop": 4,
                  }
                }
              >
                Property 2
              </Styled(Text)>
              <Styled(Text)
                alert={true}
                rt={true}
                status={true}
                style={
                  Object {
                    "marginTop": 0,
                  }
                }
              >
                WOO 30 YEARS AGO
              </Styled(Text)>
            </ColumnNB>
          </Styled(CardItem)>
        </TouchableOpacity>
      </HomepassCarousel>
    </Styled(ListItem)>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Activity
        </Styled(Text)>
      </Styled(Body)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
    >
      <Styled(Feed)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(Feed)>
    </Styled(ListItem)>
  </Styled(Content)>
</Styled(Container)>
`;
